airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -315,31 +319,33 @@ private static boolean isUpgradingThroughVersionBreak(final AirbyteVersion airby
 
   @Deprecated
   @SuppressWarnings({"DeprecatedIsStillUsed"})
-  private static Optional<AirbyteVersion> runFileMigration(final AirbyteVersion airbyteVersion,
-                                                           final ConfigRepository configRepository,
-                                                           final ConfigPersistence seed,
-                                                           final SpecFetcher specFetcher,
-                                                           final JobPersistence jobPersistence,
-                                                           final Configs configs)
+  private static AirbyteVersion runFileMigration(final AirbyteVersion airbyteVersion,
+                                                 final AirbyteVersion initialAirbyteDatabaseVersion,
+                                                 final ConfigRepository configRepository,
+                                                 final ConfigPersistence seed,
+                                                 final SpecFetcher specFetcher,
+                                                 final JobPersistence jobPersistence,
+                                                 final Configs configs)
       throws IOException {
     // required before migration
     // TODO: remove this specFetcherFn logic once file migrations are deprecated
     configRepository.setSpecFetcher(dockerImage -> Exceptions.toRuntime(() -> specFetcher.getSpec(dockerImage)));
 
-    Optional<AirbyteVersion> airbyteDatabaseVersion = jobPersistence.getVersion().map(AirbyteVersion::new);
-    if (airbyteDatabaseVersion.isPresent() && isDatabaseVersionBehindAppVersion(airbyteVersion, airbyteDatabaseVersion.get())) {
+    // version in the database after migration is run.
+    AirbyteVersion airbyteDatabaseVersion = null;
```

Comment:
It may be slightly simpler to just set this to `initialAirbyteDatabaseVersion` here instead of `null`. Then then the last line can just be `return airbyteDatabaseVersion;`

I don't feel super strongly though so up to you